src/ListView/ListView.js

```diff
@@ -0,0 +1,109 @@
+import cx from 'classnames'
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import ListViewActions from './ListViewActions'
+import ListViewAdditionalInfo from './ListViewAdditionalInfo'
+import ListViewBody from './ListViewBody'
+import ListViewCheckbox from './ListViewCheckbox'
+import ListViewDescription from './ListViewDescription'
+import ListViewDescriptionHeading from './ListViewDescriptionHeading'
+import ListViewDescriptionText from './ListViewDescriptionText'
+import ListViewExpand from './ListViewExpand'
+import ListViewGroupItem from './ListViewGroupItem'
+import ListViewGroupItemContainer from './ListViewGroupItemContainer'
+import ListViewGroupItemHeader from './ListViewGroupItemHeader'
+import ListViewIcon from './ListViewIcon'
+import ListViewInfoItem from './ListViewInfoItem'
+import ListViewItem from './ListViewItem'
+import ListViewLeft from './ListViewLeft'
+import ListViewMainInfo from './ListViewMainInfo'
+import ListViewRow from './ListViewRow'
+
+/**
+ * Components in this module are used as building blocks for ListViewItem and
+ * ListViewRow. If needed, components can be used to create custom ListViewItem
+ *
+ * Custom ListView example:
+ *
+ * <ListView>
+ *   <ListViewGroupItem stacked expanded>
+ *     <ListViewGroupItemHeader toggleExpanded={functionToToggle}> // required only if the ListViewGroupItem is supposed to be expandable
+ *       <ListViewExpand expanded />
+ *       <ListViewCheckbox />
+ *       <ListViewActions>
+ *         // buttons, dropdowns...
+ *       </ListViewActions>
+ *       <ListViewMainInfo>
+ *         <ListViewLeft>
+ *           <ListViewIcon size="sm" name={iconName} />
+ *         </ListViewLeft>
+ *         <ListViewBody>
+ *           <ListViewDescription>
+ *             <ListViewDescriptionHeading>
+ *               {name}
+ *             </ListViewDescriptionHeading>
+ *             <ListViewDescriptionText>
+ *               {description}
+ *             </ListViewDescriptionText>
+ *           </ListViewDescription>
+ *           <ListViewAdditionalInfo>
+ *             <ListViewInfoItem>
+ *               <ListViewIcon type="pf" name="flavor" />
+ *               {Item1}
+ *             </ListViewInfoItem>
+ *             <ListViewInfoItem>
+ *               <ListViewIcon type="pf" name="cpu" />
+ *               {Item2}
+ *             </ListViewInfoItem>
+ *           </ListViewAdditionalInfo>
+ *         </ListViewBody>
+ *       </ListViewMainInfo>
+ *     </ListViewGroupItemHeader>
+ *
+ *     <ListViewGroupItemContainer onClose={functionWhichClosesMe} expanded>
+ *       <Row>Some content goes here</Row>
+ *     </ListViewGroupItemContainer>
+ *
+ *   </ListViewGroupItem>
+ *   ...
+ * </ListView>
+ */
+
+/**
+ * ListView component wraps ListViewItems
+ */
+const ListView = ({ children, className, ...rest }) => {
+  const classes = cx('list-group list-view-pf list-view-pf-view', className)
```

Comment:
Patternfly is moving away from the list-view-pf implementation and suggesting the use of the list-pf implementation. Have you considered this?

Reply:
Yes! I have started #63 to introduce the new list pattern. My understanding is that this existing pattern is in use in some other areas today still (and that we can likely remove this list view in the future). This PR kind of expanded when we discovered that `ListGroup` names in React-Bootstrap conflicted with the existing class names used. I think this PR can more generally be purposed to introduce `List-Group`, `Icon`, and the other basic components we can extend from React-Boostrap. Ideally we can start using these components throughout. Would also like to merge #77 soon. 

thoughts on this @jeff-phillips-18 ?

Reply:
OK, makes sense. Thanks.

Reply:
Do we have the fact that we are moving away from the list-view-pf implementation documented somewhere?  Should we add comments to this code to tell people they should consider using the list-pf components instead?

Reply:
+1. I'm all for this. Should we wait until #77 is merged? My guess is we will have to support both for a bit. I am awaiting API feedback on #77 before adding the other examples, but am happy to hand that over if someone else gets to it first. I'd just love getting the basics in this one and #77 out of the way. I think we are having some repo organization meetings next week that should help with this!

Reply:
Sounds good.  It will be good to document this somewhere though, I am probably going to use list-pf instead of list-view-pf for stuff in my vertical nav, only because I saw this PR.